pdp/static/js/crmp_controls.js

```diff
@@ -128,6 +128,11 @@ function getResetButton(map) {
     $(reset_button).bind('click', function () {
         this.form.reset();
         filter_clear(map);
+        // This wipes the date filters so the call to dateChange gets an empty string and
+        //throws ugly errors that eventually leads to erroneous tile loading from WMS.
+        map.filters.date = date_filter('YYYY/MM/DD','YYYY/MM/DD');
+        $('#from-date')[0].value=map.filters.date.filters[0].value
+        $('#to-date')[0].value=map.filters.date.filters[1].value
```

Comment:
Let's put this into `filter_clear()` instead of here. `filter_clear` is only called in one place, here, but I think this logic more rightly belongs inside it, not outside it. That function should be revised to look like:
```
function filter_clear(map) {
    var date = date_filter('YYYY/MM/DD','YYYY/MM/DD');
    map.filters = {
        values: map.filters.values,
        date: date
    };
    $('#from-date')[0].value = date.filters[0].value;
    $('#to-date')[0].value = date.filters[1].value;
    map.getLayersByName("Polygon selection")[0].removeAllFeatures();
}
```

I don't particularly like mixing the setting of `map.filters` with the manipulation of the DOM (`$(...)`), but the PDP code mixes concerns all the time and I think this is the right place for it.

Reply:
Thanks for looking through this. I don't know much about javascript and even less about how the DOM works, but, I couldn't come up with another way, in my hacking around, to force populate the date range text boxes so that the page looked like it does at start.

That brings up a potential nuclear and inelegant solution which would be to force a reload of the page when the reset filters button is pressed...

Reply:
I think your solution is likely fine. My only quibble was with how the code is organized. 

Did you run this changed code to check it out?

Reply:
No, I did all my debugging initially through the running application on firefox. Inserted breakpoints in the code then forced the changes to occur in the console. I don't think it would be wise for me to try to get this running on my laptop as I know I would spend hours getting it all configured... :grimacing:

Reply:
running your change in the js console in firefox gives the desired result. I put a breakpoint at crmp_filters.js:152, hit "Reset Filters" on the crmp app and when the execution stops at the break, paste in your changes and then resume the execution and behaviour is as desired.

So, not a proper test, but seems to be strong support that it is working.